### PR TITLE
Add useSteppingAff hook

### PR DESCRIPTION
**Pull request by jship**
This PR adds a `useSteppingAff` variant of the `useAff` hook that preserves the previous async effect's result up until the next async effect's run completes. We use this variant at Lumi in cases where we'd like to avoid showing a page load indicator when refetching data from the backend, as the stepping behavior allows us to show the previous value right up until we've finished the API calls.

### CODE REVIEW

**Review by megamaddu — APPROVED** [review]

File: src/React/Basic/Hooks/Aff.purs
Comment (ptrfrncsmrph) [inline src/React/Basic/Hooks/Aff.purs:63]: Do you think it would make sense to export `useAff'` as well? Contrived example, but I could imagine maybe wanting something other than `identity` or `const Nothing` if wanting to track loading state along with the `Aff`'s result (to preserve and render the previous result but disable a button while some request is in flight, e.g.)
```haskell
useReloadingAff
  :: forall deps a
   . Eq deps
  => deps
  -> Aff a
  -> Hook
       (UseAff deps { isLoading :: Boolean, result :: a })
       (Maybe { isLoading :: Boolean, result :: a })
useReloadingAff deps aff =
  React.useAff' initUpdater deps aff'
  where
  initUpdater = (map <<< map) (_ { isLoading = true })
  aff' = map { isLoading: false, result: _ } aff
```

Comment (megamaddu) [reply src/React/Basic/Hooks/Aff.purs:63]: I'd want to have a pretty convincing use case. It's easy enough to copy/paste for current edge cases, so I worry it's not worth exposing that complexity.

Comment (megamaddu) [reply src/React/Basic/Hooks/Aff.purs:63]: But happy to defer to others since I'm not an active user right now.

Same goes for merging/publishing this.

Comment (ptrfrncsmrph) [reply src/React/Basic/Hooks/Aff.purs:63]: > It's easy enough to copy/paste for current edge cases, so I worry it's not worth exposing that complexity.

Keeping the external API simple makes sense to me 👍 In cases where I've wanted anything other than the `Maybe a` state that `useAff` returns I'd just use a separate `useState` and use `useAff` for scheduling. So for the snippet I had above I think I'd probably just do
```haskell
useReloadingAff deps aff = React.do
  state /\ setState <- useState Nothing
  useAff deps do
    liftEffect $ setState (map (_ { isLoading = true }))
    result <- aff
    liftEffect $ setState \_ -> Just { isLoading: false, result }
  pure state
```
which is not much more complicated than the version using `useAff'`.

I'm not really an active user either aside from side projects. Lumi's use case for `useStepperAff` seems like enough motivation for merging this as-is, but I'll leave this open for feedback for a couple more days before merging.

**Pull request merged**
